modules/internal/ChapelBase.chpl

```diff
@@ -636,17 +636,73 @@ module ChapelBase {
   //
   // left and right shift on primitive types
   //
-  inline proc <<(a: int(?w), b: integral) return __primitive("<<", a, b);
-  inline proc <<(a: uint(?w), b: integral) return __primitive("<<", a, b);
 
-  inline proc >>(a: int(?w), b: integral) return __primitive(">>", a, b);
-  inline proc >>(a: uint(?w), b: integral) return __primitive(">>", a, b);
+  inline proc bitshiftChecks(a, b: integral) {
+    use HaltWrappers;
+
+    if b < 0 {
+      var msg = "Cannot bitshift " + a:string + " by " + b:string +
+                " because " + b:string + " is less than 0";
+      HaltWrappers.boundsCheckHalt(msg);
+    } else if b >= numBits(a.type) {
+      var msg = "Cannot bitshift " + a:string + " by " + b:string +
+                " because " + b:string + " is >= the bitwidth of " +
+                a.type:string;
+      HaltWrappers.boundsCheckHalt(msg);
+    }
+  }
+
+  inline proc bitshiftChecks(param a, param b: integral) {
+    param hdr = "Cannot bitshift " + a:string + " by " + b:string +
+                " because ";
```

Comment:
it'd be nice if both param and non-param versions have the same approach for creating the error message